apis/management.cattle.io/v3/catalog_types.go

```diff
@@ -24,6 +24,9 @@ type CatalogSpec struct {
 	URL         string `json:"url,omitempty" norman:"required"`
 	Branch      string `json:"branch,omitempty"`
 	CatalogKind string `json:"catalogKind,omitempty"`
+	Username    string `json:"username,omitempty"`
+	Password    string `json:"password,omitempty" norman:"type=password"`
+	Private     bool   `json:"private,omitempty" norman:"default=false"`
```

Comment:
do we need the field `private`? wouldn't having non empty username/password imply the catalog is private?